backend/api/Services/MissionSchedulingService.cs

```diff
@@ -452,32 +455,8 @@ private async Task StartMissionRun(MissionRun queuedMissionRun)
             return ongoingMissions;
         }
 
-        private async Task<bool> TheSystemIsAvailableToRunAMission(string robotId, string missionRunId)
+        private bool TheSystemIsAvailableToRunAMission(Robot robot, MissionRun missionRun)
         {
-            bool ongoingMission = await OngoingMission(robotId);
```

Comment:
Instead of checking the database again here (which can cause concurrency issues which are hard to keep track of), we instead rely on always being able to handle failing to schedule a mission due to the robot being busy.